File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gs: List[int], cs: List[int]) -> int:
        gsa, csa = 0, 0
        ans, psum =0, 0
        for i in range(len(cs)):
            gsa+=gs[i]
            csa+=cs[i]
            psum+= (gs[i]-cs[i])
            if psum<0:
                psum=0
                ans = i+1
        if csa>gsa:
            return -1
        return ans

File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gs, vector<int>& cs) {
        if(accumulate(begin(gs),end(gs),0ll)<accumulate(begin(cs),end(cs),0ll))
            return -1;
        int ans =0;
        int temp=0;
        for(int i=0;i<cs.size();++i){
            temp+= (-cs[i]+gs[i]);
            if(temp<0){
                ans=i+1;
                temp=0;
            }
        }
        return ans;
    }
};